packages/lib-react-components/src/Markdownz/Markdownz.spec.js

```diff
@@ -11,4 +13,257 @@ describe('<Markdownz />', function () {
   })
 
   it('renders without crashing', function () { })
+
+  it('renders markdown', function () {
+    expect(wrapper.find('h1')).to.have.lengthOf(1)
+    expect(wrapper.html().includes('Zooniverse')).to.be.true
+  })
+
+  describe('#buildResourceURL', function () {
+    let buildResourceURLSpy
+    before(function () {
+      buildResourceURLSpy = sinon.spy(Markdownz.prototype, 'buildResourceURL')
+    })
+
+    afterEach(function () {
+      buildResourceURLSpy.resetHistory()
+    })
+
+    after(function () {
+      buildResourceURLSpy.restore()
+    })
+
+    it('should return an empty string if symbol is not defined', function () {
+      const url = wrapper.instance().buildResourceURL('srallen')
+      expect(url).to.equal('')
+    })
+
+    it('should return an empty string if the resource is falsy, but the mention symbol is', function () {
+      const url = wrapper.instance().buildResourceURL('', '@')
+      expect(url).to.equal('')
+    })
+
+    describe('when the symbol is #', function () {
+      it('should return the expected url with a search query param', function () {
+        const resource = 'tigers'
+        const baseURL = ''
+        const expectedReturnValue = `${baseURL}/talk/search?query=${resource}`
+        wrapper.instance().buildResourceURL(resource, '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+      })
+
+      it('should return a url with the project slug if props.projectSlug is defined', function () {
+        const resource = 'tigers'
+        const projectSlug = 'zooniverse/snapshot-wakanda'
+        const expectedReturnValue = `/projects/${projectSlug}/talk/tag/${resource}`
+        wrapper.setProps({ projectSlug })
+        wrapper.instance().buildResourceURL('tigers', '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+        wrapper.setProps({ projectSlug: '' })
+      })
+
+      it('should return a url with props.baseURI if defined', function () {
+        const resource = 'tigers'
+        const baseURI = 'https://classrooms.zooniverse.org'
+        const expectedReturnValue = `${baseURI}/talk/search?query=${resource}`
+        wrapper.setProps({ baseURI })
+        wrapper.instance().buildResourceURL('tigers', '#')
+        expect(buildResourceURLSpy).to.have.been.calledOnce
+        expect(buildResourceURLSpy).to.have.returned(expectedReturnValue)
+        wrapper.setProps({ baseURI: '' })
+      })
+    })
+
+    describe('when the symbol is @', function () {
+      it('should return a url for the user profile', function () {
+        const resource = 'srallen'
```

Comment:
Should we test for usernames that include non-alphabetic characters? I think `am.zooni` has had problems being @-mentioned in the past.